File: first architecture AWS lambda/DDI-first-architecture-js/Injector.js
const AWS = require("aws-sdk")

const tableName = "service-registry2";

class Injector {

    constructor(){
        if (!Injector.instance){
            this.connection = new AWS.DynamoDB.DocumentClient();
            Injector._instance = this
        }
        return Injector._instance
    }


        async registerService(id, serviceName, serviceUrl) {
            
            try{
                await this.connection.put(
                    {
                        TableName: tableName,
                        Item: {
                            id: id,
                            ServiceName: serviceName,
                            ServiceAddress: serviceUrl,
                        }
                    }
                ).promise()
                } catch (error){
                    console.log(`error during service registration: ${error}`)
                }

        }
    
        async getService(id) {
           
            let result
            const start = Date.now()
            try{
                result = await this.connection.get(
                    {
                        TableName: tableName,
                        Key: {
                            id: id,
                        }
                    }
                ).promise()

                const end = Date.now()
                console.log(`Read from DynamoDB table executed in ${end-start} ms `)
                return result.Item
            } catch (error){
                console.log(`error during service retrieving: ${error}`)
            }
          
        }

  
}

module.exports=Injector;

File: first architecture AWS lambda/DDI-first-architecture-go/pkg/injector/Injector.go
package injector

import (
	"fmt"
	"log"
	"sync"
	"time"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
)

const tableName = "service-registry2"

type Injector struct {
	connection *dynamodb.DynamoDB
}

type Service struct {
	id             string `dynamodbav:"id"`
	ServiceName    string `dynamodbav:"ServiceName"`
	ServiceAddress string `dynamodbav:"ServiceAddress"`
}

var instance *Injector
var once sync.Once

func GetInstance() *Injector {
	once.Do(func() {
		sess := session.Must(session.NewSessionWithOptions(session.Options{
			SharedConfigState: session.SharedConfigEnable,
		}))
		i := &Injector{
			connection: dynamodb.New(sess),
		}
		instance = i
	})

	return instance
}

func (i *Injector) RegisterService(id, serviceName, serviceURL string) {
	_, err := i.connection.PutItem(&dynamodb.PutItemInput{
		TableName: aws.String(tableName),
		Item: map[string]*dynamodb.AttributeValue{
			"id":             {S: aws.String(id)},
			"ServiceName":    {S: aws.String(serviceName)},
			"ServiceAddress": {S: aws.String(serviceURL)},
		},
	})
	if err != nil {
		fmt.Printf("error during service registration")
	}
}

func (i *Injector) GetService(id string) Service {
	start := time.Now().UnixMilli()
	result, _ := i.connection.GetItem(&dynamodb.GetItemInput{
		TableName: aws.String(tableName),
		Key: map[string]*dynamodb.AttributeValue{
			"id": {S: aws.String(id)},
		},
	})
	end := time.Now().UnixMilli()

	log.Printf("Read from DynamoDB table executed in %d ms", (end - start))

	service := Service{}
	dynamodbattribute.UnmarshalMap(result.Item, &service)

	return service
}

